From 5a98b48bc4609e1e484dd49699d3ef20c24af55c Mon Sep 17 00:00:00 2001
From: Alain Volmat <alain.volmat@foss.st.com>
Date: Sat, 23 Mar 2024 21:44:40 +0100
Subject: [PATCH 1035/1141] media: dcmipp: pixelproc: correct typo in
 set_selection

Correct a typo within the set_selection handling to have height
properly handled.

Change-Id: I160faf0a66fea6dee0737b1407c1b972f18dad1a
Signed-off-by: Alain Volmat <alain.volmat@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/369268
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Philippe CORNU <philippe.cornu@foss.st.com>
Domain-Review: Philippe CORNU <philippe.cornu@foss.st.com>
---
 drivers/media/platform/st/stm32/stm32-dcmipp/dcmipp-pixelproc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/media/platform/st/stm32/stm32-dcmipp/dcmipp-pixelproc.c
+++ b/drivers/media/platform/st/stm32/stm32-dcmipp/dcmipp-pixelproc.c
@@ -589,7 +589,7 @@ static int dcmipp_pixelproc_set_selectio
 		if (s->r.height > crop->height)
 			s->r.height = crop->height;
 		else if (s->r.height < (crop->height / DCMIPP_MAX_DOWNSCALE_RATIO))
-			s->r.height = crop->width / DCMIPP_MAX_DOWNSCALE_RATIO;
+			s->r.height = crop->height / DCMIPP_MAX_DOWNSCALE_RATIO;
 		s->r.top = 0;
 		s->r.left = 0;
 
